src/v/cluster/read_replica_manager.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cloud_storage/remote.h"
+#include "s3/client.h"
+
+namespace cluster {
+
+// on every core
+class read_replica_manager {
```

Comment:
nit: the name is a bit misleading
the object of this class doesn't manages read-replicas, it only used to create them

Reply:
also, probably there is no need to create a `sharded_service<read_repliica_manager>`
you can simply create it ad-hoc, it's almost free to create since it accepts only on reference in c-tor, this can probably simplify things a bit

Reply:
changed code, not creating `sharded_service<read_replica_manager>` now

re naming: I was following the pattern here where we call "manager" anything we don't have better name for [wiki page](https://vectorizedio.atlassian.net/wiki/spaces/CORE/pages/156958746/Controller+Internals#Subsystem-design-pattern)